Lib/test/test_free_threading/test_set.py

```diff
@@ -0,0 +1,43 @@
+import unittest
+
+from threading import Thread, Barrier
+from unittest import TestCase
+
+from test.support import threading_helper
+
+
+@threading_helper.requires_working_threading()
+class TestSet(TestCase):
+    def test_repr_clear(self):
+        """Test repr() of a set while another thread is calling clear()"""
+        NUM_ITERS = 10
+        NUM_REPR_THREADS = 10
+        barrier = Barrier(NUM_REPR_THREADS + 1)
+        s = {1, 2, 3, 4, 5, 6, 7, 8}
+        set_reprs = []
+
+        def clear_set():
+            barrier.wait()
+            s.clear()
+
+        def repr_set():
+            nonlocal set_repr
+            barrier.wait()
+            set_reprs.append(repr(s))
+
+        for _ in range(NUM_ITERS):
+            set_reprs = []
+            threads = [Thread(target=clear_set)]
+            for _ in range(NUM_REPR_THREADS):
+                threads.append(Thread(target=repr_set))
+            for t in threads:
+                t.start()
+            for t in threads:
+                t.join()
+
+            for set_repr in set_reprs:
+                self.assertIn(set_repr, ("set()", "{1, 2, 3, 4, 5, 6, 7, 8}"))
```

Comment:
... this would report all failing set reprs, not just the first one, and avoid hardcoding the repr we're expecting. (4/4)

```suggestion
            self.assertEqual(set_reprs, expected)
```

Reply:
This assertion ended up not working:

expected is `{ "set()", "{1, 2, 3, 4, 5, 6, 7, 8}" }`

set_reprs may be any of the three cases:

* `{ "set()", "{1, 2, 3, 4, 5, 6, 7, 8}" }`
* `{ "{1, 2, 3, 4, 5, 6, 7, 8}" }`
* `{ "set()" }`

So we could do `self.assertTrue(set_reprs.issubset(expected))` or something similar, but at that point I'd prefer the original assertion failure message.